checks/evaluation/license.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2021 Security Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package evaluation
+
+import (
+	"fmt"
+
+	"github.com/ossf/scorecard/v4/checker"
+	sce "github.com/ossf/scorecard/v4/errors"
+)
+
+// License applies the score policy for the License check.
+func License(name string, dl checker.DetailLogger,
+	r *checker.LicenseData,
+) checker.CheckResult {
+	if r == nil {
+		e := sce.WithMessage(sce.ErrScorecardInternal, "empty raw data")
+		return checker.CreateRuntimeErrorResult(name, e)
+	}
+	// We expect a single license.
+	if len(r.Files) > 1 {
+		e := sce.WithMessage(sce.ErrScorecardInternal, fmt.Sprintf("invalid number of results: %d",
+			len(r.Files)))
+		return checker.CreateRuntimeErrorResult(name, e)
+	}
```

Comment:
Do we really care that strongly if some repos have >1 license files? Why not just log all of them?

Reply:
Basically, is it possible that some repos have different license requirements for different parts of the code (per-directory basis let's say). So instead of returning a runtime error maybe we should handle this more gracefully?

Reply:
the code never returns more than one license file even where there are several licenses.  Code in raw folder uses `fileparser.OnAllFilesDo(c.RepoClient, isLicenseFile, &path)` where path is a string, not an array. So the case where > 1 license in this file should never happen.

Let me know if I mis-understood something

Reply:
I see. So:

1) It should still not be creating a runtime error because we are baking in the assumption about how `raw` package is implemented into the `evaluation` package. The `evaluation` package works on `rawData` and should only handle that.
2) Maybe we should consider expanding the `rawData` logic to return >1 license files if there is possibility of this happening? Not needed in this PR, can consider for future change.

Reply:
> I see. So:
> 
> 1. It should still not be creating a runtime error because we are baking in the assumption about how `raw` package is implemented into the `evaluation` package. The `evaluation` package works on `rawData` and should only handle that.

What would you like to return? We do similar sanity checking for security policy, for example. I think what you're saying is that we should get rid of this?